### PR TITLE
Modify documentation.

### LINKED ISSUE #32: Todo list for documentation
- [x] Create a new section called "Base Types and Methods" which includes NormalFormGame.
- [x] Add random game to the section "Base Types and Methods".
- [x] Change `auto_doc_gen.jl`, so that each section does not have subsections for files, but just has subsections of "Exported" and "Internal".
- [x] [Add a `User Guide`](https://github.com/QuantEcon/Games.jl/pull/27#issuecomment-287337136) before Library. 
- [x] Add a description to docs/README.md about how to use auto_doc_gen.jl.

### PR DESCRIPTION
**Pull request by shizejin**
close #32.

Thanks very much for the suggestions of @oyamad and @sglyon!

Preview on my forked repo [here](https://shizejin.github.io/Games.jl/latest/index.html).

### CODE REVIEW

File: docs/src/index.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+```@example 1
+pure_nash(g)
+```
+
+Please see [lectures](@ref lectures) on QuantEcon for more details.
+
+## [Lectures](@id lectures)
+
+Some lectures of this package is available on [QuantEcon](https://lectures.quantecon.org):
```
Comment (oyamad) [inline docs/src/index.md]: Better to say "notebooks" instead of "lectures".

**Comment by oyamad** [comment]
Thanks! Looks great.

File: docs/src/index.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
+# Games.jl
+
+*Games.jl* is a [`Julia`](http://www.julialang.org) package for computation of Game Theory.
```
Comment (QBatista) [inline docs/src/index.md]: *Games.jl* is a [`Julia`](http://www.julialang.org) package for **Game Theory related computations**.

File: docs/src/index.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+## Installation
+
+*Games.jl* is an unregistered package that currently under development.
```
Comment (QBatista) [inline docs/src/index.md]: *Games.jl* is an unregistered package that **is** currently under development.

File: docs/src/index.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+using Games
+```
+
+The Base type `Player` can be created by passing a payoff matrix.
+
+```@example 1
+player1 = Player([3 1; 0 2])
+```
+
+A 2-player `NormalFormGame` can be created either by passing `Player` instances.
```
Comment (QBatista) [inline docs/src/index.md]: A 2-player `NormalFormGame` can be created either by passing `Player` instances**,**

File: docs/src/index.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+```@example 1
+player2 = Player([2 0; 1 3])
+g = NormalFormGame((player1, player2))
+```
+
+or by passing a payoff bimatrix directly.
```
Comment (QBatista) [inline docs/src/index.md]: or by passing a payoff **matrix** directly.
(to keep consistent with line 23)

File: docs/src/index.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+```@example 1
+payoff_bimatrix = Array(Int, 2, 2, 2)
+payoff_bimatrix[1, 1, :] = [3, 2]
+payoff_bimatrix[1, 2, :] = [1, 1]
+payoff_bimatrix[2, 1, :] = [0, 0]
+payoff_bimatrix[2, 2, :] = [2, 3]
+g = NormalFormGame(payoff_bimatrix)
+```
+
+After construction of `NormalFormGame`, we can find its Nash Equilibria by using methods of `Games`, for example, `pure_nash` finds all pure action Nash Equilibria by enumeration.
```
Comment (QBatista) [inline docs/src/index.md]: After **constructing a** `NormalFormGame`, we can find its Nash Equilibria by using methods **in** `Games`**. For** example, `pure_nash` **is a method that** finds all pure action Nash Equilibria by enumeration.

File: docs/src/index.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+```@example 1
+pure_nash(g)
+```
+
+Please see [notebooks](@ref notebooks) on QuantEcon for more details.
```
Comment (QBatista) [inline docs/src/index.md]: Please see **the** [notebooks](@ref notebooks) on QuantEcon for more details.

File: docs/src/index.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+## [Notebooks](@id notebooks)
+
+Some notebooks of this package is available on [QuantEcon](https://lectures.quantecon.org):
```
Comment (QBatista) [inline docs/src/index.md]: Some notebooks **for** this package **are** available on [QuantEcon](https://lectures.quantecon.org)**. In particular, see:**

File: docs/README.md [hunk 9]
```diff
@@ -1,20 +1,56 @@
...
+To generate documentation for `Games.jl`, run
+
 ```
 $julia make.jl
 ```
 
-The generated docs can be accessed in `build/index.html`.
+The generated docs can be accessed in `docs/build/index.html`.
+
+It includes three parts to generate the documentation.
```
Comment (QBatista) [inline docs/README.md]: Not sure what "it" refers to here -- I'm guessing the process of generating documentation?
If so:
The process of generating documentation includes three parts.

File: docs/README.md [hunk 10]
```diff
@@ -1,20 +1,56 @@
...
+
+### auto_doc_gen.jl
+
+Main part of documentation generation. It reads `src/Games.jl` to find
+out all the source files, and then write markdown files for each with
+the structure instructed by `docs/build/Structure`, which will be utilized
+by "Documenter.jl" later.
+
+Especially, pattern of each page could be modified by changing the
```
Comment (QBatista) [inline docs/README.md]: **Note that the** pattern of each page **can** be modified by changing the

File: docs/README.md [hunk 11]
```diff
@@ -1,20 +1,56 @@
...
+strings in `auto_doc_gen.jl`, called `file_page` and `section_page`.
```
Comment (QBatista) [inline docs/README.md]: strings in `auto_doc_gen.jl` called `file_page` and `section_page`.
(no comma)

File: docs/README.md [hunk 12]
```diff
@@ -1,20 +1,56 @@
...
+
+### manually written files
 
-You may modify `Structure` to arrange the structure of the website to be generated. For example:
+Two files should be written manually:
```
Comment (QBatista) [inline docs/README.md]: Two files **must** be written manually:
(which files?)

File: docs/README.md [hunk 13]
```diff
@@ -1,20 +1,56 @@
...
-Repeated Game: repeated_game_util, repeated_game
+Order: Base Types and Methods, Computing Nash Equilibria, Repeated Games
+Base Types and Methods: normal_form_game, random
+Computing Nash Equilibria: pure_nash
+Repeated Games: repeated_game_util, repeated_game
 ```
 
-The first line sets the order of pages. You can add a file name if you want it to generate a single page, or the name of section you want to create. In the following you need to declare which files are included in each section. Note that you shouldn't put a comma or dot at the end of each line.
+The first line sets the order of pages. You can add a file name if you want it to
+generate a single page, or the name of section you want to create. In the following
```
Comment (QBatista) [inline docs/README.md]: or the name of **the** section you want to create.

File: docs/README.md [hunk 14]
```diff
@@ -1,20 +1,56 @@
...
+you need to declare which files are included in each section. Note that you shouldn't
```
Comment (QBatista) [inline docs/README.md]: Note that you **should not**

File: docs/README.md [hunk 15]
```diff
@@ -1,20 +1,56 @@
...
+put a comma or dot at the end of each line.
+
+It is not necessary to declare structure for every file included in the Module.
```
Comment (QBatista) [inline docs/README.md]: It is not necessary to declare `Structure` for every file included in the Module.

File: docs/README.md [hunk 16]
```diff
@@ -1,20 +1,56 @@
...
+Files not mentioned in `Structure` will simply generate a single page automatically.
+
+#### Homepage
+
+The Homepage is written in `/docs/src/index.md`. Please keep the last part in Homepage,
+called `Library Outline`, being empty. It will be filled by `auto_doc_gen.jl` automatically.
```
Comment (QBatista) [inline docs/README.md]: called `Library Outline`, empty.
(being is not needed)

File: docs/README.md [hunk 17]
```diff
@@ -1,20 +1,56 @@
...
+
+### make.jl
 
-It is not necessary to declare structure for every file included in the Module. Files not mentioned in `Structure` will simply generate a single page automatically.
+This file includes information for "Documenter.jl" to generate documentation.
+Details [here](https://juliadocs.github.io/Documenter.jl/stable/man/guide.html#Output-formats-1)
+for how to use it.
```
Comment (QBatista) [inline docs/README.md]: Details **on how to use it can be found** [here](https://juliadocs.github.io/Documenter.jl/stable/man/guide.html#Output-formats-1).

File: docs/README.md (hunk 15 above)
Comment (shizejin) [inline docs/README.md]: I think I am referring to structure literally here, but not the file `Structure`.

File: docs/README.md (hunk 12 above)
Comment (shizejin) [inline docs/README.md]: The names of two files are showed behind as subsections, and I think it's explicit enough to not show them in this line.
These two files can be left without editing, and a documentation page will be generated anyway without structure and content on homepage. So I think maybe "should" is more appropriate here.

File: docs/README.md (hunk 15 above)
Comment (QBatista) [inline docs/README.md]: Ok, then I think "It is not necessary to declare **the** structure **of** every file included in the Module." would be good?

File: docs/README.md (hunk 12 above)
Comment (QBatista) [inline docs/README.md]: I see, I think "should" is a bit unclear here about whether you can or must do it. Instead, I would say:
"While not necessary, it is (strongly?) recommended to manually specify the two following files:"

**Comment by shizejin** [comment]
This is ready to be merged.

**Comment by oyamad** [comment]
Let me merge this as is. And then let's discuss for further improvements, if any, in a separate issue.

**Pull request merged**